aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ResourceModelHelper.java

```diff
@@ -31,13 +32,20 @@ public static boolean shouldUpdateAfterCreate(final ResourceModel model) {
                                 StringUtils.hasValue(model.getMonitoringRoleArn()) ||
                                 Optional.ofNullable(model.getBackupRetentionPeriod()).orElse(0) > 0 ||
                                 Optional.ofNullable(model.getMonitoringInterval()).orElse(0) > 0 ||
-                                isStorageParametersModified(model) ||
+                                (isSqlServer(model) && isStorageParametersModified(model)) ||
                                 BooleanUtils.isTrue(model.getManageMasterUserPassword()) ||
                                 BooleanUtils.isTrue(model.getDeletionProtection()) ||
                                 BooleanUtils.isTrue(model.getEnablePerformanceInsights())
                 );
     }
 
+    public static boolean isSqlServer(final ResourceModel model) {
+        final String engine = model.getEngine();
+        // treat unknown engines as SQLServer
```

Comment:
From my understanding, the detection of an unknown engine as sqlServer will only occur in the UpdateHandler. The `fetchEngine` method will either fill in the engine or fail hard if it is unknown. In my opinion, I would prefer to fail hard and provide a message instructing to add the Engine parameter to the template. Unfortunately, this change is not backward compatible. So, my suggestion is to log this event so that we can track it and potentially make changes in the future.

To achieve this, we can use the logger to log the following message:
```
        requestLogger.log("Unknown engine converted to SqlServer detected",
                "This indicates that we could not extract the dbinstance engine from the given customer template, and we treated it as SqlServer for backward compatibility.");
```

In order to have better accessibility for the RequestLogger, a change has been introduced in the following pull request: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/459/files. To utilize this change, you will need to pass the RequestLogger as a parameter to the method, as the logger is only given to the ResourceHandler during execution.

Reply:
This method is not accessed at all in UpdateHandler. Let's consider this enhancement in the logging improvement request.

Reply:
Could you please provide more details on how the engine will remain unknown even after the `fetchEngine` method is called at the beginning of the CreateHandler?